### DISCUSSION TITLE
How to expand nested function?

### DISCUSSION THREAD

**Discussion by MAGNETO903** [josdejong/mathjs#2670] [Q&A] [answered]
I have 2 simple functions, for example:
`f1(x) = 2*x`
`f2(x) = x^2`

And one complex function:
`f3(x) = f1(f2(x))`

Can I using mathjs expand function `f3`?
I need to get something like this:

`math.expand("f3(x)") // return 2*x^2`

or 

`math.simplify(math.parse("f3(x)")) // should return 2*x^2 `

Can I do this?

**Comment by josdejong** [comment, marked as answer]
That is an interesting one. There is no functionality built in for this, but I think you could build it quite easily.

After parsing an expression like `f1(x)`, you could transform the parsed expression and replace all occurrences of `f1` with it's implementation `2*x` using the method `.transform(...)`. When replacing, you should make sure the function variables should be correctly renamed too when needed, such that `f1(y)` will be replaced with `2*y`.

See docs: https://mathjs.org/docs/expressions/expression_trees.html
